doc/concepts/concepts.rst

```diff
@@ -1,13 +1,227 @@
+
+.. image:: /_static/images/planning_scene_monitor.svg
+
+World Geometry Monitor
+----------------------
+
+The world geometry monitor builds world geometry using information from the sensors on the robot such as LIDARs or depth cameras and from user input. It uses the ``occupancy map monitor`` described below to build a 3D representation of the environment around the robot and augments that with information on the ``planning_scene`` topic for adding object information.
+
+3D Perception
+-------------
+
+3D perception in MoveIt is handled by the ``occupancy map monitor``. The occupancy map monitor uses a plugin architecture to handle different kinds of sensor input as shown in the Figure above. In particular, MoveIt has inbuilt support for handling two kinds of inputs:
+
+- **Point clouds**: handled by the ``point cloud occupancy map updater`` plugin
+
+- **Depth images**: handled by the ``depth image occupancy map updater`` plugin
+
+Note that you can add your own types of updaters as a plugin to the occupancy map monitor.
+
+Octomap
+-------
+
+The Occupancy map monitor uses an `Octomap <http://octomap.github.io/>`_ to maintain the occupancy map of the environment. The Octomap can actually encode probabilistic information about individual cells although this information is not currently used in MoveIt. The Octomap can directly be passed into FCL, the collision checking library that MoveIt uses.
+
+Depth Image Occupancy Map Updater
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The depth image occupancy map updater includes its own *self-filter*, i.e. it will remove visible parts of the robot from the depth map. It uses current information about the robot (the robot state) to carry out this operation.
+
+.. raw:: html
+
+   <hr>
+
+**********
+Kinematics
+**********
+
+The Kinematics Plugin
+---------------------
+
+MoveIt uses a plugin infrastructure, especially targeted towards allowing users to write their own inverse kinematics algorithms. Forward kinematics and finding jacobians is integrated within the RobotState class itself. The default inverse kinematics plugin for MoveIt is configured using the `KDL <https://github.com/orocos/orocos_kinematics_dynamics>`_ numerical jacobian-based solver. This plugin is automatically configured by the MoveIt Setup Assistant.
+
+******************
+Collision Checking
+******************
+
+Collision checking in MoveIt is configured inside a Planning Scene using the _CollisionWorld_ object. Fortunately, MoveIt is setup so that users never really have to worry about how collision checking is happening. Collision checking in MoveIt is mainly carried out using the `FCL <https://flexible-collision-library.github.io/>`_ package - the primary collision checking library of MoveIt.
```

Comment:
```suggestion
Collision checking in MoveIt is configured inside a Planning Scene using the _CollisionWorld_ object. Fortunately, MoveIt is set up so that users never really have to worry about how collision checking is happening. Collision checking in MoveIt is mainly carried out using the `FCL <https://flexible-collision-library.github.io/>`_ package - the primary collision checking library of MoveIt.
```